### PR TITLE
Use nbsite to generate documentation website

### PR DESCRIPTION
**Pull request by ceball**
screen shot 2018-07-24 at 12 38 31 pm

screen shot 2018-07-24 at 12 38 53 pm

Ready to review (but not merge).

To do:
  * [x] reference manual
  * [x] deal with 'historical release notes' on sidebar
  * [x] depends on new release of nbsite (same as https://github.com/pyviz/pyviz/pull/81 and https://github.com/bokeh/datashader/pull/619)
  * [x] get travis to build site correctly
  * [ ] update the conditions for build/deploy

There are lots of improvements to make to the docs, but those can come after this PR.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -1,73 +1,39 @@
...
+_NAV =  (
+    ('API', 'Reference_Manual/param'),
+    ('About', 'About'),
+)
+
+html_context.update({
+    'PROJECT': project,
+    'DESCRIPTION': description,
+    'AUTHOR': authors,
+    # will work without this - for canonical (so can ignore when building locally or test deploying)
```
Comment (jbednar) [inline doc/conf.py]: for canonical what?

File: examples/About.ipynb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Param is part of PyViz, a collaboration between the maintainers of [Bokeh](http://bokeh.pydata.org), [HoloViews](http://holoviews.org), [GeoViews](http://geo.holoviews.org), [Datashader](https://github.com/bokeh/datashader), and Param, including James A. Bednar, Philipp Rudiger, Jean-Luc Stevens, and Bryan Van de Ven.\n",
```
Comment (jbednar) [inline examples/About.ipynb]: and Chris Ball?

**Comment by jlstevens** [comment]
Looking forward to seeing the new site!

File: doc/conf.py (hunk 1 above)
Comment (ceball) [reply doc/conf.py]: I guess you're prompting me to clarify this in the text, but in case not: https://support.google.com/webmasters/answer/139066

Comment (jbednar) [reply doc/conf.py]: Just saying that it's a parse error; "for <adjective>" isn't meaningful. :-)  Maybe:

```
# Ensures that all pages have a single canonical URL; can ignore for local builds
```

**Comment by jbednar** [comment]
Looks good.  Well, it looks quite plain, but it's good to see it come in line with the other sites, and I'm sure we'll be able to pretty it up a bit.

**Comment by ceball** [comment]
dev website's at https://ioam-docs.github.io/param-dev/ (can be updated with new dev tag or with `website_dev` tag).

Next release, the param.pyviz.org website should automatically switch too.

**Pull request merged**